cpp/include/raft/sparse/mst/detail/mst_kernels.cuh

```diff
@@ -109,23 +109,38 @@ __global__ void min_edge_per_supervertex(
   const weight_t* altered_weights, vertex_t* temp_src, vertex_t* temp_dst,
   weight_t* temp_weights, const weight_t* min_edge_color, const vertex_t v) {
   auto tid = get_1D_idx<vertex_t>();
-
   if (tid < v) {
     vertex_t vertex_color_idx = color_index[tid];
     vertex_t vertex_color = color[vertex_color_idx];
     edge_t edge_idx = new_mst_edge[tid];
-
     // check if valid outgoing edge was found
     // find minimum edge is same as minimum edge of whole supervertex
     // if yes, that is part of mst
     if (edge_idx != std::numeric_limits<edge_t>::max()) {
       weight_t vertex_weight = altered_weights[edge_idx];
       if (min_edge_color[vertex_color] == vertex_weight) {
-        temp_src[tid] = tid;
-        temp_dst[tid] = indices[edge_idx];
-        temp_weights[tid] = weights[edge_idx];
-
-        mst_edge[edge_idx] = true;
+        auto dst = indices[edge_idx];
+        auto dst_edge_idx = new_mst_edge[dst];
+        auto dst_color = color[color_index[dst]];
+        // vertices added each other
+        // only if destination has found an edge
+        // the edge points back to source
+        // the edge is minimum edge found for dst color
+        if (dst_edge_idx != std::numeric_limits<edge_t>::max() &&
+            indices[dst_edge_idx] == tid &&
+            min_edge_color[dst_color] == altered_weights[dst_edge_idx]) {
+          if (vertex_color < dst_color) {
+            temp_src[tid] = tid;
```

Comment:
Can you break this logic out with a boolean to avoid code repetition?

Reply:
I believe this was a change you had provided. If you can provide an example of the change you'd like to see, I don't mind adding it to this PR.